src/cursor/cur_hs.c

```diff
@@ -142,6 +142,50 @@ __wt_hs_cursor_search_near(WT_SESSION_IMPL *session, WT_CURSOR *cursor, int *exa
     return (ret);
 }
 
+/*
+ * __curhs_next --
+ *     WT_CURSOR->next method for the hs cursor type.
+ */
+static int
+__curhs_next(WT_CURSOR *cursor)
+{
+    WT_CURSOR *file_cursor;
+    WT_CURSOR_HS *hs_cursor;
+    WT_DECL_RET;
+    WT_SESSION_IMPL *session;
+
+    hs_cursor = (WT_CURSOR_HS *)cursor;
+    file_cursor = hs_cursor->file_cursor;
+    CURSOR_API_CALL(cursor, session, next, CUR2BT(file_cursor));
```

Comment:
We have to consider the btree_id, key, timestamp set for hs_cursor as part of set key. We need to return WT_NOTFOUND based on the criteria set (i.e. if only btree_id is set, we need to return WT_NOTFOUND when we are crossing over to a new btree in history store.). For reference you can check __curhs_prev_visible/__curhs_next_visible

Reply:
Fixed.